model-providers/openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureRestApiFilterResolver.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkiverse.langchain4j.azure.openai;
+
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkiverse.langchain4j.auth.ModelAuthProvider;
+import io.quarkiverse.langchain4j.openai.common.OpenAiRestApi;
+import io.quarkiverse.langchain4j.openai.common.RestApiFilterResolver;
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+
+public class AzureRestApiFilterResolver implements RestApiFilterResolver {
```

Comment:
I believe it has to become a synthetic CDI bean - you can see an example of how such beans are created in AzureOpenAI recorder - and you'd have the config properties available to the function that creates it